### PR TITLE
feat: nestjs user api and e2e tests (#256)

### PR DESCRIPTION
**Pull request by dorukucak**
Fixed: #256 

### Description
Created nestjs server and implemented a user API that expose an /register endpoint.

**Resolves**: # (#256)

---

### Changes Made
- [ ] Changes in **`apps`** folder (specify the app and briefly describe the changes):
  - [ ] `Web`
  - [ ] `Native`
  - [x] `Api`

- [ ] Changes in **`packages`** folder (specify the package and briefly describe the changes):
  - [ ] `Core`

---

### Type of Change
- [ ] 🐛 **Bug fix** (non-breaking change which fixes an issue)
- [x] ✨ **New feature** (non-breaking change which adds functionality)
- [] 💥 **Breaking change** (fix or feature that would cause existing functionality to not work as expected)
- [ ] 📝 **Documentation update** (changes)

---

## Screenshots
|       Before        |       After        |
| :-----------------: | :----------------: |
| "screenshot before" | "screenshot after" |
---

### How Has This Been Tested?
- [x] Supertest Jest integration
- [x] Jest unit tests

---

### Checklist:
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

---

### Additional Comments
*(Optional) Add any additional comments or notes for reviewers here.*

### CODE REVIEW

File: apps/user/libs/utils/swagger.ts [hunk 1]
```diff
@@ -0,0 +1,13 @@
+import { INestApplication } from '@nestjs/common';
+import { DocumentBuilder, SwaggerModule } from '@nestjs/swagger';
+
+export function setupSwagger(app: INestApplication): void {
```
Comment (dadiorchen) [inline apps/user/libs/utils/swagger.ts:4]: Awesome you are using swagger, can you give the guide to access it? In the Readme?

File: apps/user/src/user/user.service.ts [hunk 2]
```diff
@@ -0,0 +1,10 @@
+import { Injectable } from '@nestjs/common';
+import { RegisterUserDto } from '@dtos/register-user.dto';
+
+@Injectable()
+export class UserService {
+  async registerUser(registerUserDto: RegisterUserDto) {
+    const username = registerUserDto.username;
+    return { message: 'create user service', data: { username } };
```
Comment (dadiorchen) [inline apps/user/src/user/user.service.ts:8]: @dorukucak this is a e2e test, so I need you to do the real thing with Keycloak.

**Comment by dadiorchen** [comment]
@dorukucak you finish it? Can you e2e test pass? Please add your command to run your e2e in the readme.md, then we can merge your code.

**Comment by dorukucak** [comment]
Working on implementing keycloak. Couldn't find a keycloak endpoint to register user. Do we have that?

**Comment by dadiorchen** [comment]
@pierrelstan we need your help here

**Comment by pierrelstan** [comment]
@dorukucak  I think these links could help find the register endpoint.:

-  www.keycloak.org/docs-api/latest/rest-api/index.html
- https://documenter.getpostman.com/view/7294517/SzmfZHnd#f3dd8450-3605-4ce8-8ba6-1e2c15161d0e

**Comment by dorukucak** [comment]
Okay so, I had to create a Keycloak service account and give it manage-user privilages. Now we can succesfully create users using the /register endpoint. However, we need to handle API secrets before we can run tests on github workflow etc.

**Comment by dorukucak** [comment]
resolved conflicts in yarn.lock

**Comment by dorukucak** [comment]
This is ready

**Comment by dorukucak** [comment]
- utilized cross-env package to set dev environment variables for now. we can change it in the future.
- rebased on top of main

**Comment by pierrelstan** [comment]
@dorukucak I noticed that the cross-env package seems to be missing in apps/user/package.json. Could you add it to ensure everything works as expected?

File: apps/user/package.json
Comment (pierrelstan) [inline apps/user/package.json:1]: @dorukucak I noticed that the cross-env package seems to be missing in apps/user/package.json. Could you add it to ensure everything works as expected?

Comment (dorukucak) [reply apps/user/package.json:1]: That's weird. Ok now added as a dev dependency.

**Review by pierrelstan — APPROVED** [review]

**Comment by dadiorchen** [comment]
> * utilized cross-env package to set dev environment variables for now. we can change it in the future.
> * rebased on top of main

Cross-env is a good practice, thanks.

**Pull request merged**